kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/executor/CommandExecutorService.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.executor;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * This interface provides methods for running system processes or executing system commands.
+ *
+ * @noimplement This interface is not intended to be implemented by clients.
+ */
+@ProviderType
+public interface CommandExecutorService {
+
+    /**
+     * Synchronously executes a system command.
+     * 
+     * @param command
+     *            the {@link Command} to be executed
+     * @return a {@link CommandStatus} object
+     */
+    public CommandStatus execute(Command command);
+
+    /**
+     * Asynchronously executes a system command.
+     * 
+     * @param command
+     *            the {@link Command} to be executed
+     * @param callback
+     *            the consumer called when the command returns
+     */
+    public void execute(Command command, Consumer<CommandStatus> callback);
+
+    /**
+     * Stops the system process identified by the given {@link Pid}.
+     * 
+     * @param pid
+     *            the {@link Pid} of the process to be stopped
+     * @param signal
+     *            the {@link Signal} send to the process to stop it. If null, a default signal will be send
+     * @return true if the stop was succeeded
+     */
+    public boolean stop(Pid pid, Signal signal);
+
+    /**
+     * Kills the system commands containing all the tokens in the given command line.
+     * If more processes are found, all of them will be killed.
+     * 
+     * @param commandLine
+     *            the command to be killed
+     * @param signal
+     *            the {@link Signal} sent to the command to kill it. If null, a default signal will be sent
+     * @return true if the kill was succeeded
```

Comment:
here and in the other return I believe that we should mention that we return a boolean and that the boolean is true in a certain condition.